### PR TITLE
[WIP] {Down,Up}loading files

### PR DESCRIPTION
**Pull request by agarciamontoro**
WIP of the implementation of the `downloadTo` function

### CODE REVIEW

**Comment by agarciamontoro** [comment]
Trying to get what we talked about in the blackboard, @eparejatobes. At the moment this does not even compile, but I need you to take a first look before I continue with this; I am not sure at all I'm following the right path here.

File: src/main/scala/code.scala [hunk 1]
```diff
@@ -4,18 +4,52 @@ import ohnosequences.loquat._
...
   val downloadTo:
-    BasespaceAPI  =>
-    File          =>
-    BasespaceURL  =>
-    BasespaceError + CheckedFile =
-    ???
+    WSClient     =>
+    BasespaceURL =>
+    FileStream   =>
+    (WSClient, FileStream, Future[BasespaceError + File]) =
```
Comment (eparejatobes) [inline src/main/scala/code.scala]: for what we want to do here returning `Future` won't help in any way. Better to get rid of it ASAP

File: src/main/scala/code.scala [hunk 2]
```diff
@@ -4,18 +4,52 @@ import ohnosequences.loquat._
...
+              response.body.runWith(
+                Sink.foreach[akka.util.ByteString] { bytes =>
+                  fileStream._2.write(bytes.toArray)
+                }
+              ) map {
+                _ => Right[BasespaceError, File](fileStream._1)
+              } recoverWith {
+                case e => Left[BasespaceError, File](BasespaceError(e.toString))
+              }
+          }
```
Comment (eparejatobes) [inline src/main/scala/code.scala]: block on this (`Await ...`)

**Comment by agarciamontoro** [comment]
Can you take a look to the current status, @eparejatobes? Thanks!

I still have to implement `checkFile` and `notifyTo` methods. I'll need some help with the latter.

**Review by eparejatobes — COMMENTED** [review]
OK. Have you tested this somehow?

**Comment by agarciamontoro** [comment]
Not yet, should we do it from the `webmiodx` app?

**Comment by eparejatobes** [comment]
> Not yet, should we do it from the `webmiodx` app?

We'll do it here.

**Comment by eparejatobes** [comment]
@agarciamontoro what about the token etc?

**Comment by agarciamontoro** [comment]
> @agarciamontoro what about the token etc?

@eparejatobes There's no need for the token anymore, as we can tell the BaseSpace API to give us S3 public URLs.

Btw, I just added some (successful) tests.

**Comment by eparejatobes** [comment]
> @eparejatobes There's no need for the token anymore, as we can tell the BaseSpace API to give us S3 public URLs.

you mean pre-signed, right? for how long they remain valid?

**Comment by agarciamontoro** [comment]
I don't know, the documentation is pretty poor. See the option `filehrefcontentresolution` in the docs for `/datasets/{id}/files`: https://developer.basespace.illumina.com/docs/content/documentation/rest-api/api-reference#spanclassoperation-nameGETspanstrongdatasetsidfilesstrong

**Pull request merged**